ignite/cli
# refactor(network): genesis parser

close #2197
close #2210

## Description

- Add the possibility to fetch a genesis from a URL inside a tarball, checking the folders recursively to find the `genesis.json` file.
- Refactor the commands to download/open the genesis file only one time
- Cache the Genesis into disk instead memory
- Create a more efficiently JSON reader, reading line by line for we can get only one parameter
- Open the Genesis file only one time
- Set the new values when writing into the file using the io.Writer method `Write([]byte) (int, error)`

#### How to test

- Publish a chain with a tarball genesis file
```shell
starport n --nightly chain publish https://github.com/Pantani/mars --genesis https://github.com/Pantani/genesis/raw/main/genesis.tar.gz --chain-id mars-1
```

- Publish a chain with a simple genesis from the URL
```shell
starport n --nightly chain publish https://github.com/Pantani/mars --genesis https://github.com/Pantani/genesis/raw/main/genesis.json  --chain-id mars-1
```

_the command should fail in the genesis verification because the genesis file is not valid for the chain_

_the CLI logs are a little bit different from the issue to avoid adding too much logic into the code_
